pkg/api/defaults.go

```diff
@@ -460,22 +460,17 @@ func (p *Properties) setAgentProfileDefaults(isUpgrade, isScale bool) {
 					if profile.OSDiskSizeGB != 0 && profile.OSDiskSizeGB < VHDDiskSizeAKS {
 						profile.Distro = Ubuntu
 					} else {
-						if profile.IsNSeriesSKU() {
-							profile.Distro = AKSDockerEngine
-						} else {
-							profile.Distro = AKS
-						}
+						profile.Distro = AKS
 					}
 				} else {
 					profile.Distro = Ubuntu
 				}
 				// Ensure distro is set properly for N Series SKUs, because
-				// (1) At present, "aks-docker-engine" and "ubuntu" are the only working distro base for running GPU workloads on N Series SKUs
-				// (2) Previous versions of aks-engine had working implementations using the "aks" distro value,
-				//     so we need to hard override it in order to produce a working cluster in upgrade/scale contexts
-			} else if p.OrchestratorProfile.IsKubernetes() && (isUpgrade || isScale) && profile.IsNSeriesSKU() {
-				if profile.Distro == AKS {
-					profile.Distro = AKSDockerEngine
+				// Previous versions of aks-engine required the docker-engine distro for N series vms,
+				// so we need to hard override it in order to produce a working cluster in upgrade/scale contexts
+			} else if p.OrchestratorProfile.IsKubernetes() && (isUpgrade || isScale) {
+				if profile.Distro == AKSDockerEngine {
+					profile.Distro = AKS
```

Comment:
this should test for 18.04 as well

Reply:
Not sure what you mean

Reply:
you can disregard my comment actually, I was thinking we needed to preserve 18.04 if the distro was AKS1804 but this is only for clusters using the AKSDockerEngine distro so my comment doesn't apply.